### PR TITLE
test: add unit tests for chat_completions translation module

### LINKED ISSUE #231: test: add unit tests for OpenAI Responses-to-ChatCompletions translation
## Context

Coverage is at 90.12%, targeting 95%. The chat completions translation module has **1,126 lines with zero test coverage**.

## Files needing tests

| File | Lines | What it does |
|------|-------|-------------|
| `apis/src/openai/translation/chat_completions.rs` | 1,126 | Translates OpenAI Responses API requests/responses to/from Chat Completions-compatible provider formats, handling messages, tools, and response formatting |

## Suggested test areas

- Request translation: Responses API input items to ChatCompletions messages (text, images, tool calls, tool results)
- Response translation: ChatCompletions responses back to Responses API format
- Tool handling: function calling translation in both directions
- Edge cases: empty inputs, malformed tool schemas, streaming vs non-streaming
- Format-specific handling: different provider response shapes

## Impact

Covering these 1,126 lines would push overall coverage from ~90% toward ~92.4%.

### PR DESCRIPTION
**Pull request by leseb**
## Summary

- Adds **64 new unit tests** (87 total) for `apis/src/openai/translation/chat_completions.rs`, covering all major code paths
- Tests span `ResponseContext` construction, request translation (messages, content parts, tools, tool_choice, text format, reasoning), and response translation (finish reasons, usage, service tier, tool calls, refusals, logprobs, full resource shape)

## Test areas covered

| Area | Tests added |
|------|------------|
| `ResponseContext` construction & defaults | 4 |
| Request: message edge cases | 5 |
| Request: content part handling | 5 |
| Request: function call serialization | 5 |
| Request: tool definitions | 3 |
| Request: tool_choice variants | 5 |
| Request: text format / reasoning | 5 |
| Response: finish reasons & status | 3 |
| Response: completed_at behavior | 2 |
| Response: empty/no choices | 4 |
| Response: content + refusal | 5 |
| Response: usage tokens | 3 |
| Response: service tier fallbacks | 3 |
| Response: tool call output items | 2 |
| Response: full resource shape & defaults | 6 |
| Response: non-object error | 1 |
| Response: message item id & annotations | 2 |

Resolves #231

## Test plan

- [x] All 87 tests pass (`cargo test --lib -p praxis-ai-apis -- translation::tests`)
- [x] No clippy warnings

### CODE REVIEW

File: apis/src/openai/translation/mod.rs [hunk 1]
```diff
@@ -671,4 +671,1177 @@ mod tests {
...
+
+        assert_eq!(ctx.completed_at, Some(20));
+    }
+
+    // -------------------------------------------------------------------------
+    // Request translation: message edge cases
+    // -------------------------------------------------------------------------
+
+    #[test]
+    fn empty_instructions_do_not_produce_system_message() {
```
Comment (aslakknutsen) [inline apis/src/openai/translation/mod.rs]: duplicates simple_inputs_map_or_drop_cleanly

File: apis/src/openai/translation/mod.rs [hunk 2]
```diff
@@ -671,4 +671,1177 @@ mod tests {
...
+    }
+
+    #[test]
+    fn untyped_item_without_role_or_content_returns_error() {
+        let error = map_error(&json!({
+            "model": "m",
+            "input": [{"foo": "bar"}]
+        }));
+
+        assert!(error.contains("unknown"));
```
Comment (aslakknutsen) [inline apis/src/openai/translation/mod.rs]: both UnsupportedInputItemType and UnsupportedContentPartType could be caught by "unknown", intentional?

**Comment by aslakknutsen** [comment]
As this PR resolves #231, I can't see any  test case for;

* edge case "malformed tool schemas"
* also the silent drop in function_call_tool_call when call_id or name  is missing seems untested

File: apis/src/openai/translation/mod.rs (hunk 1 above)
Comment (leseb) [reply apis/src/openai/translation/mod.rs]: true, but harmless, i'll remove it anyway :)

File: apis/src/openai/translation/mod.rs [hunk 3]
```diff
@@ -671,4 +671,1177 @@ mod tests {
             response["choices"][0]["logprobs"]["content"]
         );
     }
+
+    // -------------------------------------------------------------------------
```
[the remark this answers is not in the document]
Comment (leseb) [reply apis/src/openai/translation/mod.rs:675]: i'll handle that with a `lint` action improvement later

File: apis/src/openai/translation/mod.rs (hunk 1 above)
Comment (leseb) [reply apis/src/openai/translation/mod.rs]: 351426491de92d0817da2c1336e35c956ffa369e

File: apis/src/openai/translation/mod.rs [hunk 4]
```diff
@@ -671,4 +671,1177 @@ mod tests {
...
+                .to_string()
+                .contains("Chat Completions response must be a JSON object")
+        );
+    }
+
+    // -------------------------------------------------------------------------
+    // Response translation: finish reasons
+    // -------------------------------------------------------------------------
+
+    fn make_response_context(request: &Value) -> super::chat_completions::ResponseContext {
```
[the remark this answers is not in the document]
Comment (leseb) [reply apis/src/openai/translation/mod.rs]: 351426491de92d0817da2c1336e35c956ffa369e

File: apis/src/openai/translation/mod.rs (hunk 2 above)
Comment (leseb) [reply apis/src/openai/translation/mod.rs]: yes this was intentional but the test assertion was too loose, updated in 351426491de92d0817da2c1336e35c956ffa369e

**Review by aslakknutsen — APPROVED** [review]
